packages/venia-concept/src/RootComponents/Category/category.js

```diff
@@ -1,119 +1,97 @@
+const Category = props => {
+    const {
+        id,
+        currentPage,
+        pageSize,
+        prevPageTotal,
+        setCurrentPage,
+        setPrevPageTotal
+    } = props;
 
-    // TODO: Should not be a default here, we just don't have
-    // the wiring in place to map route info down the tree (yet)
-    static defaultProps = {
-        id: 3
+    const pageControl = {
+        currentPage: currentPage,
+        setPage: setCurrentPage,
+        updateTotalPages: setPrevPageTotal,
+        totalPages: prevPageTotal
     };
 
-    componentDidUpdate(prevProps) {
-        // If the current page has changed, scroll back up to the top.
-        if (this.props.currentPage !== prevProps.currentPage) {
-            window.scrollTo(0, 0);
-        }
-    }
-
-    render() {
-        const {
-            id,
-            classes,
-            currentPage,
-            pageSize,
-            prevPageTotal,
-            setCurrentPage,
-            setPrevPageTotal
-        } = this.props;
+    const [queryResult, queryApi] = useQuery(categoryQuery);
+    const { data, error, loading } = queryResult;
+    const { runQuery, setLoading } = queryApi;
+    const classes = mergeClasses(defaultClasses, props.classes);
 
-        const pageControl = {
-            currentPage: currentPage,
-            setPage: setCurrentPage,
-            updateTotalPages: setPrevPageTotal,
-            totalPages: prevPageTotal
-        };
+    useEffect(() => {
+        setLoading(true);
+        runQuery({
+            variables: {
+                id: Number(id),
+                onServer: false,
+                pageSize: Number(pageSize),
+                currentPage: Number(currentPage)
+            }
+        });
+    }, [id, pageSize, currentPage, runQuery, setLoading]);
 
-        return (
-            <Query
-                query={categoryQuery}
-                variables={{
-                    id: Number(id),
-                    onServer: false,
-                    pageSize: Number(pageSize),
-                    currentPage: Number(currentPage)
-                }}
-            >
-                {({ loading, error, data }) => {
-                    if (error) return <div>Data Fetch Error</div>;
-                    // If our pagination component has mounted, then we have
-                    // a total page count in the store, so we continue to render
-                    // with our last known total
-                    if (loading)
-                        return pageControl.totalPages ? (
-                            <CategoryContent
-                                pageControl={pageControl}
-                                pageSize={pageSize}
-                            />
-                        ) : (
-                            loadingIndicator
-                        );
+    useEffect(() => {
+        window.scrollTo({
+            left: 0,
+            top: 0,
+            behavior: 'smooth'
+        });
+    }, [currentPage]);
```

Comment:
Great use of `useEffect()` here. Window scrolling is absolutely a side effect, and we only want to do it when the `currentPage` changes. This could eventually be a candidate for a utility hook.

Reply:
It's also correct to do this at this point in the component, before any early returns. 👍

Reply:
Should we add `pageSize` to the dependent property list? I think we should scroll to top when that changes.
```suggestion
    }, [currentPage, pageSize]);
```
Even the `id` property might be a good idea; it would be cool to reuse most of the UI when you navigate between entities of the same type.

Reply:
Moved this logic into the hook above, that has all these same dependencies.